### DISCUSSION TITLE
Planning a plasma hack week for 2021

### DISCUSSION THREAD

**Discussion by namurphy** [PlasmaPy/PlasmaPy#983] [General]
An idea that came up in discussions after last year's DPP software ecosystem mini-conference is to organize a **Plasma Hack Week** later in 2021 (probably late spring or summer).  We're hoping to recruit volunteers to be on the organizing committee so that we can hold a hack week later this spring or over the summer.  I'm hoping the organizing committee will include both prospective participants who are new to research software engineering, as well as people who can represent established projects.  A goal of Plasma Hack Week will be to introduce plasma scientists to collaborative software development practices, and help us move towards community development of research software.

This idea is akin to [Astro Hack Week](http://astrohackweek.org/2020/).  Astro Hack Week partially serves as a summer school, with tutorials and exercises in the mornings.  The afternoons are entirely unstructured to give time for unconference sessions and a hackathon.  Participants have the option of coming in with a project in mind, joining an existing project, or taking the time to learn and apply new skills.

Plasma Hack Week will need to be entirely remote in 2021, which will simplify some logistics, but complicate others.  We could keep the same structure with morning tutorials and an afternoon hackathon. However, given how exhausting it is to spend hours in a video conference, a possibility would be to have ~1–2 tutorials per week in the ~3 weeks before the actual event.  The tutorials could be open to everyone and recorded.

If you're interested in helping to organize this, or even if you're interested in participating, please either post here or email me at namurphy@cfa.harvard.edu.

Thanks!
—Nick

**Comment by rocco8773** [comment]
@namurphy Are you planning to send an email to people that participated in the mini-conf?  I know @smithsp was particularly interested in this.

Also, we should probably plan this as a virtual hack week.  Hopefully things swill be getting back to normal during summer, but I doubt many people will be traveling yet.

**Reply by namurphy** [reply]
> @namurphy Are you planning to send an email to people that participated in the mini-conf?

Yep!  I was hoping we could disseminate this more widely in the near future.

**Comment by smithsp** [comment]
A mention is sufficient.  Count me in :-)  We should make sure that we do this hack week after SULI assignments have been made/accepted.  Also, looking at the DIII-D schedule, there are maintenance weeks scheduled for Feb 9-19, Mar 29-Apr 9, May 10-21 (although partial overlap with IAEA FEC), and then this year's longer vent should start by at least Jun 28, lasting until Sep 30.  Let me know what I can do to help.

**Comment by qudsiramiz** [comment]
Hello,
I am not sure how much I can help with this, but if I can in any capacity I
will be happy to help.
--

Regards

Ramiz A. Qudsi

Department of Physics and Astronomy
University of Delaware
Newark, DE-19716

Ph: (1) 302-831-6277

**Comment by StanczakDominik** [comment]
Definitely count me in as well :)

**Comment by alfkoehn** [comment]
I'd definitely like to participate although I will very likely be more on the learning side.

**Comment by namurphy** [comment]
Thank you all for expressing interest!  I was hoping to have two initial discussions on the hack week during the weeks of February 15 and 22.  Please put your availability in this [scheduling poll](https://www.when2meet.com/?10978067-v89Vl) by Wednesday, February 10 at 19 UTC if you would like to participate in these initial planning discussions.  (If you were interested in participating but not planning the hack week, we'll make sure to post here once we have some more details.)

Some of the initial discussion topics could be: choosing a code of conduct, possible tutorial topics, the logistics of having a virtual hack week, which projects will be participating, the roles of participating projects, possible months for the hack week, and how we can build a sense of community that continues to last after the event.  A hope of mine is that this event will be participant-driven.  For example, any participant should feel free to organize a tutorial or unconference session that they hope will happen.

**Comment by namurphy** [comment]
The results of the scheduling poll are in, and we'll be having our first two discussions on planning a Plasma Hack Week on **February 19 and 26 (both Fridays) at 17:00 UTC (12 pm ET / 9 am PT / 6 pm CET)** at this [Zoom link](https://harvard.zoom.us/j/91908342677?pwd=c0p4bjdCTS9Fc1BYelpUZVFVR2Jqdz09) (passcode 227174).   

I created a google doc that we can use for our [agenda and minutes](https://docs.google.com/document/d/1hvyTB4McoUhLB49_oWT5Acr1CpcAZ41Zg7HeVQKH4ZM/edit?usp=sharing).  Please feel free to add items to the agenda that you think we should talk about. We'll also take notes during the meeting for anyone who isn't able to attend.  I've also created a [brainstorming document](https://docs.google.com/document/d/1WdSSupRpH8ufYmCkpmxkIZuVlq3tSq15xFtk_nynxlQ/edit?usp=sharing) that we can use as a place to keep track of thoughts, ideas, questions, concerns, and resources.  You are welcome to add to that document too, especially if you end up needing to miss one of the meetings.  

Thank you!  I'm looking forward to it! 😸

**Comment by namurphy** [comment]
From the discussion today, the plan is to do this the week of June 28 to July 3, as long as this does not overlap with the SULI summer school.  The tutorials would be happening the week before, and be recorded to accommodate time zones.

**Comment by namurphy** [comment]
Thanks to @StanczakDominik, we now have a [Plasma Coders Discord channel](https://discord.gg/X6qHyCpy).  We'll use this for the hack week (including planning it) as well as after to keep the community going.

We'll be having our next meeting on next **Friday, March 12** at 17 UTC (6 pm CET / 12 pm ET / 9 am PT) at this [Zoom link](https://harvard.zoom.us/j/91908342677?pwd=c0p4bjdCTS9Fc1BYelpUZVFVR2Jqdz09) (passcode 227174).  Here are the links to the [agenda and minutes](https://docs.google.com/document/d/1hvyTB4McoUhLB49_oWT5Acr1CpcAZ41Zg7HeVQKH4ZM/edit?usp=sharing) and the shared [brainstorming document](https://docs.google.com/document/d/1WdSSupRpH8ufYmCkpmxkIZuVlq3tSq15xFtk_nynxlQ/edit?usp=sharing).
